### ISSUE TITLE
T-SQL: Explicit aliasing of columns broken when using =

### ISSUE THREAD

**Issue by fdw** [sqlfluff/sqlfluff#2749]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I fear #2727 broke something.
Take this example:
```sql
select key = value
from table
```

With L012 set to explicit aliases, the result is invalid SQL:
```sql
select as key = value
from table
```

Instead, I would expect
```sql
select value as key
from table
```
or (preferably)
```sql
select key = value
from table
```

### Expected Behaviour

See above: Not broken code, either using `as` or `=` but not both ;)

### Observed Behaviour

`sqlfluff` currently adds an `as`, but also keeps the aliasing with `=`.

### How to reproduce

See above: Use that snippet and call `sqlfluff fix` on it.

### Dialect

T-SQL

### Version

Current `master`

### Configuration

```
[sqlfluff]
dialect = tsql

[sqlfluff:rules:L011] # Aliasing
aliasing = implicit

[sqlfluff:rules:L012] # Aliasing
aliasing = explicit
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Yes I really think L012 is currently incompatible with this usage of aliasing. So think it needs to be turned off.

But concerning it corrupts SQL if people don't realise that. Would be good to update L012 to ignore this aliasing before 0.11.0 to avoid that issue.

**Comment by tunetheweb** [comment]
I get linting errors with your examples. `key`, `value` and `table` are not valid sql.

**Comment by tunetheweb** [comment]
Managed to fix it in #2750

**Comment by fdw** [comment]
Awesome, thank you!
